src/plugins/lua/bindings/utils.cpp

```diff
@@ -215,6 +218,37 @@ void setupUtilsModule()
                 return FilePath::fromString(path);
             };
 
+            utils.new_enum("OpenModeFlag",
+                "NotOpen", QIODeviceBase::NotOpen,
+                "ReadOnly", QIODeviceBase::ReadOnly,
+                "WriteOnly", QIODeviceBase::WriteOnly,
+                "ReadWrite", QIODeviceBase::ReadWrite,
+                "Append", QIODeviceBase::Append,
+                "Truncate", QIODeviceBase::Truncate,
+                "Text", QIODeviceBase::Text,
+                "Unbuffered", QIODeviceBase::Unbuffered,
+                "NewOnly", QIODeviceBase::NewOnly,
+                "ExistingOnly", QIODeviceBase::ExistingOnly
+            );
+
+            utils.new_usertype<FileSaver>(
+                "FileSaver",
+                sol::meta_function::construct,
+                sol::factories(
+                     [](const FilePath &path, QIODeviceBase::OpenModeFlag mode) -> std::shared_ptr<FileSaver> {
+                        return std::make_shared<FileSaver>(path, mode);
+                     },
+                     [](sol::object, const FilePath &path, QIODeviceBase::OpenModeFlag mode) -> std::shared_ptr<FileSaver> {
+                        return std::make_shared<FileSaver>(path, mode);
+                     }),
+                "finalize", [](FileSaver& fs){
+                    return fs.finalize();
+                },
```

Comment:
I think you don't need lambda here. It should be possible to do:

```suggestion
                "finalize",
                &FileSaver::finalize,
```

Reply:
It doesn't work for some reason. Sol refuses to compile it that way.